### PR TITLE
add new strip dQdx MEs

### PR DESCRIPTION
**Pull request by mtosi**
as well described by Bill in one of his recent presentation
https://indico.cern.ch/event/342236/session/5/contribution/10/material/slides/0.pdf

a new quantity, dQ/dx, will be used by the tracking in order to mitigate the effect of the PU
it is therefore necessary to have this new quantity w/in the list of the strip MEs

dqdx_fromorigin_pu25
this plot comes from runTheMatrix -l 25202.0 [TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15+MINIAODMCUP15]

### CODE REVIEW

**Comment by mtosi** [comment]
@wmtford @deguio @threus @venturia @benelli @cerati @rovere @VinInn

File: DataFormats/SiStripCluster/interface/SiStripClusterTools.h [hunk 1]
```diff
@@ -29,7 +29,7 @@ namespace siStripClusterTools {
 
   template<typename Clus>
   float chargePerCM(DetId detid, Clus const & cl) {
```
Comment (slava77) [inline DataFormats/SiStripCluster/interface/SiStripClusterTools.h]: was it never instantiated/never used previously? 
(else how could it compile)

Comment (VinInn) [inline DataFormats/SiStripCluster/interface/SiStripClusterTools.h]: it is ill defined...
Clus was supposed to be a Cluster not a ClusterRef
@mtosi  please use 
```
siStripClusterTools::chargePerCM(detid, *cluster, LV);
```
and do not modify this template function...
I may have to add some SFINAE trick

**Comment by mtosi** [comment]
ciao,
is this PR fine, now ?
@deguio @VinInn @slava77 @cerati @rovere @danduggan @threus @venturia @wmtford

**Comment by VinInn** [comment]
ok for me

**Comment by deguio** [comment]
+1
to be tested

**Comment by mtosi** [comment]
I've just realized there was a typo in 2 MEs name,
not they are fixed

sorry for the inconvenience

**Comment by mtosi** [comment]
weird ... when I tested my changes I ran runTheMatrix -l 25202.0 w/o any problems ...
I'm re-doing the test, now

**Comment by slava77** [comment]
ok for me, especially that now the reco signature is not required.
IB used for testing appears to be bad, I guess tests will be rerun later.

On 1/12/15 2:50 AM, mia tosi wrote:
> ciao,
> is this PR fine, now ?
> @deguio <https://github.com/deguio> @VinInn <https://github.com/VinInn>
> @slava77 <https://github.com/slava77> @cerati
> <https://github.com/cerati> @rovere <https://github.com/rovere>
> @danduggan <https://github.com/danduggan> @threus
> <https://github.com/threus> @venturia <https://github.com/venturia>
> @wmtford <https://github.com/wmtford>
> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/7110#issuecomment-69541627>.
> 

-- 
***********************************************************************
 Vyacheslav (Slava) Krutelyov
 TAMU: Physics Dept Texas A&M MS4242, College Station, TX 77843-4242
 CERN: 42-R-027
 AIM/Skype: siava16              googleTalk: slava77@gmail.com
 (630) 291-5128  Cell (US)       +41 76 275 7116 Cell (CERN)

-----------------------------------------------------------------------

**Comment by deguio** [comment]
+1

**Comment by deguio** [comment]
it seems that the error is not related

**Comment by deguio** [comment]
please test

**Comment by mtosi** [comment]
as I wrote 2 days ago,
I re-did the test running runTheMatrix -l 25202.0 and it worked

please, if there are really issues w/ this PR, let me know
it is kind of puzzling what you are reporting ....

**Comment by mtosi** [comment]
ping @deguio

**Comment by deguio** [comment]
the crash is not related.
a more recent IB should be used for testing.

F.

**Comment by deguio** [comment]
+1

File: DQM/SiStripCommon/src/SiStripFolderOrganizer.cc [hunk 2]
```diff
@@ -327,6 +385,7 @@ void SiStripFolderOrganizer::setLayerFolder(uint32_t rawdetid, const TrackerTopo
   }
 
   lokal_folder += rest.str();
+  std::cout << "[SiStripFolderOrganizer::setLayerFolder] lokal_folder: " << lokal_folder << "(used by dbe_->setCurrentFolder)" << std::endl;
```
Comment (ktf) [inline DQM/SiStripCommon/src/SiStripFolderOrganizer.cc]: This is causing a bunch of printouts! Can it please be cleaned up?

**Comment by mtosi** [comment]
I'm very sorry, I thought I already cleared all of them
I've just committed the new version

**Pull request merged**